homework/4/assets/js/scripts.js

```diff
@@ -0,0 +1,356 @@
+		this.dataObj = dataObj;
+		this.id = dataObj.id;
+		this.text = dataObj.text;
+		this.date = new Date(dataObj.dateCreated).toISOString();
+
+		this.$root = null;
+		this.$editBtn = null;
+		this.$deleteBtn = null;
+	}
+
+	getView() {
+		return (`
+			<li class="message-list_item message-field" data-id="${this.id}" data-mode="default">
+				<span class="message-field_time">${this.date}</span>
+				<div class="message-field_">
+					<p class="message-field_p">${this.text}</p>
+					<textarea class="message-field_textarea"></textarea>
+				</div>
+				<div class="message-field_controls">
+					<button class="btn message-field_save">Save</button>
+					<button class="btn message-field_edit">Edit</button>
+					<button class="btn message-field_delete">Delete</button>
+				</div>
+			</li>
+		`);
+	}
+
+	render($container) {
+		$container.insertAdjacentHTML('beforeend', this.getView());
+
+		this.getFields($container);
+		
+		this.$editBtn.addEventListener('click', e => {
+			e.preventDefault();
+
+			this.$root.dataset.mode = 'edit';
+			this.$textarea.value = this.$p.textContent;
+		});
+
+		this.$saveBtn.addEventListener('click', e => {
+			e.preventDefault();
+
+			const editEvent = new CustomEvent('editMessage', { detail: {
+				id: this.id,
+				text: this.$textarea.value
+			} });
+			this.$root.dispatchEvent(editEvent);
+
+			this.$root.dataset.mode = 'default';
+		});
+
+		this.$deleteBtn.addEventListener('click', e => {
+			e.preventDefault();
+
+			const deleteEvent = new CustomEvent('deleteMessage', { detail: { id: this.id } });
+			this.$root.dispatchEvent(deleteEvent);
+		});
+	}
+
+	getFields($container) {
+		this.$root = $container.querySelector(`[data-id="${this.id}"]`);
+		this.$p = this.$root.querySelector('.message-field_p');
+		this.$textarea = this.$root.querySelector('.message-field_textarea');
+		this.$editBtn = this.$root.querySelector('.message-field_edit');
+		this.$saveBtn = this.$root.querySelector('.message-field_save');
+		this.$deleteBtn = this.$root.querySelector('.message-field_delete');
+	}
+
+	setText(text) {
+		this.$p.innerText = text;
+	}
+}
+
+class Settings {
+	constructor() {
+		this.$root = document.querySelector('.messanger_settings');
+		this.$form = this.$root.querySelector('.settings_form');
+		this.$trigger = document.getElementById('settings-trigger');
+		this.$saveBtn = document.getElementById('settings-submit');
+
+		// Form fields
+		this.$sortValueSelect = document.getElementById('settings-sort-value');
+		this.$sortDirSelect = document.getElementById('settings-sort-dir');
+		this.$limit = document.getElementById('settings-limit');
+		this.$skip = document.getElementById('settings-skip');
+		
+		this.values = {
+			sort: {
+				value: null,
+				dir: null
+			},
+			limit: null,
+			skip: null
+		};
+
+		this.init();
+	}
+
+	init() {
+		this.getSortValues();
+		this.getDefaults();
+		this.addListeners();
+	}
+
+	getSortValues() {
+		fetch('http://localhost:3000/messages/fields')
```

Comment:
```suggestion
		fetch(`${window.location.origin}:3000/messages/fields`)
```
Чтобы не хардкодить)